payments/src/lib.rs

```diff
@@ -0,0 +1,578 @@
+		/// Payment request was completed by sender
+		PaymentRequestCompleted { from: T::AccountId, to: T::AccountId },
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The selected payment does not exist
+		InvalidPayment,
+		/// The selected payment cannot be released
+		PaymentAlreadyReleased,
+		/// The selected payment already exists and is in process
+		PaymentAlreadyInProcess,
+		/// Action permitted only for whitelisted users
+		InvalidAction,
+		/// Payment is in review state and cannot be modified
+		PaymentNeedsReview,
+		/// Unexpeted math error
+		MathError,
+		/// Payment request has not been created
+		RefundNotRequested,
+		/// Dispute period has not passed
+		DisputePeriodNotPassed,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// This allows any user to create a new payment, that releases only to
+		/// specified recipient The only action is to store the details of this
+		/// payment in storage and reserve the specified amount.
+		#[transactional]
+		#[pallet::weight(T::WeightInfo::pay())]
+		pub fn pay(
+			origin: OriginFor<T>,
+			recipient: T::AccountId,
+			asset: AssetIdOf<T>,
+			amount: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+			// create PaymentDetail and add to storage
+			let payment_detail = <Self as PaymentHandler<T>>::create_payment(
+				who.clone(),
+				recipient.clone(),
+				asset,
+				amount,
+				PaymentState::Created,
+				T::IncentivePercentage::get(),
+				None,
+			)?;
+			// reserve funds for payment
+			<Self as PaymentHandler<T>>::reserve_payment_amount(&who, &recipient, payment_detail)?;
+			// emit paymentcreated event
+			Self::deposit_event(Event::PaymentCreated {
+				from: who,
+				asset,
+				amount,
+			});
+			Ok(().into())
+		}
+
+		/// This allows any user to create a new payment with the option to add
+		/// a remark, this remark can then be used to run custom logic and
+		/// trigger alternate payment flows. the specified amount.
+		#[transactional]
+		#[pallet::weight(T::WeightInfo::pay_with_remark(remark.len().try_into().unwrap_or(T::MaxRemarkLength::get())))]
+		pub fn pay_with_remark(
+			origin: OriginFor<T>,
+			recipient: T::AccountId,
+			asset: AssetIdOf<T>,
+			amount: BalanceOf<T>,
+			remark: BoundedDataOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			// create PaymentDetail and add to storage
+			let payment_detail = <Self as PaymentHandler<T>>::create_payment(
+				who.clone(),
+				recipient.clone(),
+				asset,
+				amount,
+				PaymentState::Created,
+				T::IncentivePercentage::get(),
+				Some(remark),
+			)?;
+			// reserve funds for payment
+			<Self as PaymentHandler<T>>::reserve_payment_amount(&who, &recipient, payment_detail)?;
+			// emit paymentcreated event
+			Self::deposit_event(Event::PaymentCreated {
+				from: who,
+				asset,
+				amount,
```

Comment:
The remark should be in the event and I don't see too much reason to have two pay. We can always use the with remark version and pass empty remark if it is not needed.